modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/database/H2TreeIndex.java

```diff
@@ -887,6 +890,32 @@ public GridH2RowMessage toSearchRowMessage(SearchRow row) {
         return res;
     }
 
+    /**
+     * Removes all index rows that belong to the specific data partitions.
+     *
+     * @param parts Partitions.
+     * @param shouldStop Allows to check stop condition.
+     */
+    public void purge(Set<Integer> parts, IgniteCallable<Boolean> shouldStop) throws IgniteCheckedException {
+        if (F.isEmpty(parts))
+            return;
+
+        if (segments.length == 1)
```

Comment:
What we gain with this optimization?
Can we simply keep the for loop to make this code more clear?